tests/components/device_tracker/test_asuswrt.py

```diff
@@ -31,77 +20,33 @@
 }}
 
 
-def setup_module():
-    """Set up the test module."""
-    global FAKEFILE
-    FAKEFILE = get_test_config_dir('fake_file')
-    with open(FAKEFILE, 'w') as out:
-        out.write(' ')
-
-
-def teardown_module():
-    """Tear down the module."""
-    os.remove(FAKEFILE)
-
-
-@pytest.mark.skip(
-    reason="These tests are performing actual failing network calls. They "
-    "need to be cleaned up before they are re-enabled. They're frequently "
-    "failing in Travis.")
-class TestComponentsDeviceTrackerASUSWRT(unittest.TestCase):
-    """Tests for the ASUSWRT device tracker platform."""
-
-    hass = None
-
-    def setup_method(self, _):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        mock_component(self.hass, 'zone')
-
-    def teardown_method(self, _):
-        """Stop everything that was started."""
-        self.hass.stop()
-        try:
-            os.remove(self.hass.config.path(device_tracker.YAML_DEVICES))
-        except FileNotFoundError:
-            pass
-
-    def test_password_or_pub_key_required(self):
-        """Test creating an AsusWRT scanner without a pass or pubkey."""
-        with assert_setup_component(0, DOMAIN):
-            assert setup_component(
-                self.hass, DOMAIN, {DOMAIN: {
-                    CONF_PLATFORM: 'asuswrt',
-                    CONF_HOST: 'fake_host',
-                    CONF_USERNAME: 'fake_user',
-                    CONF_PROTOCOL: 'ssh'
-                }})
-
-    @mock.patch(
-        'homeassistant.components.device_tracker.asuswrt.AsusWrtDeviceScanner',
-        return_value=mock.MagicMock())
-    def test_get_scanner_with_password_no_pubkey(self, asuswrt_mock):
-        """Test creating an AsusWRT scanner with a password and no pubkey."""
-        conf_dict = {
-            DOMAIN: {
+async def test_password_or_pub_key_required(hass):
+    """Test creating an AsusWRT scanner without a pass or pubkey."""
+    with MockDependency('aioasuswrt') as mocked_asus:
+        mocked_asus.async_get_connected_devices = mock_coro_func()
+        hass.data[DATA_ASUSWRT] = mocked_asus
+        scanner = await async_get_scanner(
```

Comment:
Don't call `async_get_scanner` directly. Set up the asuswrt component with a config and let it set up the device tracker platform. That way, the config will be validated properly in the test.

Platforms should basically never be set up directly in tests. Always set up the corresponding component.